src/contentmanager.cpp

```diff
@@ -575,6 +575,7 @@ void ContentManager::updateLibrary() {
 #define CATALOG_URL "library.kiwix.org"
 void ContentManager::updateRemoteLibrary(const QString& content) {
     QtConcurrent::run([=]() {
+        QMutexLocker locker(&remoteLibraryLocker);
```

Comment:
This lock protect against to update of the remote library in the same time.
But it doesn't protect from a usage of the library (by the UI/ContentManager for example) while we are updating the library.
We should put the (same) lock also when we want to access `m_remoteLibrary`.

Reply:
Fair point. Done it.